### PR TITLE
Gauge and count axis movement

### PR DESCRIPTION
**Pull request by gardleopard**
This adds up all axis movements in prints.

test by pulling the plugin from url: https://github.com/gardleopard/OctoPrint-Prometheus-Exporter/archive/axis_travel_metrics.zip

### CODE REVIEW

File: octoprint_prometheus_exporter/__init__.py [hunk 1]
```diff
@@ -155,6 +161,7 @@ def on_event(self, event, payload):
 			# reset the extrusion counter
 			self.parser.reset()
 			self.last_extrusion_counter = 0
+			self.last_x_travel = 0
```
Comment (tg44) [inline octoprint_prometheus_exporter/__init__.py]: don't we need to reset all the 3 last travel variables here?

File: octoprint_prometheus_exporter/gcodeparser.py [hunk 2]
```diff
@@ -90,19 +93,31 @@ def parse_fan_speed(self, line):
 
         return None
 
+    def process_axis_movement(self, target_position, current_position):
```
Comment (tg44) [inline octoprint_prometheus_exporter/gcodeparser.py]: Don't we have abs? like `abs(target-currennt)`

File: octoprint_prometheus_exporter/__init__.py [hunk 3]
```diff
@@ -175,6 +182,18 @@ def on_event(self, event, payload):
 	extrusion_total = Counter('octoprint_extrusion_total', 'Filament extruded total')
 	extrusion_print = Gauge('octoprint_extrusion_print', 'Filament extruded this print')
 
+	# X TRAVEL
```
Comment (tg44) [inline octoprint_prometheus_exporter/__init__.py]: We have a hook function, which is fired after a print is finished/failed/canceled. I think we should reset, or unsubscribe the travel_print variables there.

**Comment by tg44** [comment]
Ohh, and also, pls add the new metrics to the readme :)

**Comment by gardleopard** [comment]
Thanks for the feedback, I will make corrections when I have some time :)

File: octoprint_prometheus_exporter/gcodeparser.py (hunk 2 above)
Comment (gardleopard) [reply octoprint_prometheus_exporter/gcodeparser.py]: Good catch, testing that.

File: octoprint_prometheus_exporter/__init__.py (hunk 3 above)
Comment (gardleopard) [reply octoprint_prometheus_exporter/__init__.py]: I followed how the extrusion counter works. I think they should all behave similar. Wdyt?

**Comment by tg44** [comment]
Thx the contrib!

**Pull request merged**